components/contactsList/index.jsx

```diff
@@ -54,19 +57,48 @@ function ContactsList() {
   const [search, setSearch] = useState("");
 
   const [addressBook, addressBookError] = useAddressBook();
-  const [people, peopleError] = usePeople(addressBook);
-  const isLoading =
-    (!addressBook && !addressBookError) || (!people && !peopleError);
+  const { data: people, error: peopleError, mutate: peopleMutate } = usePeople(
+    addressBook
+  );
+  const profiles = useProfiles(people);
+
+  const {
+    session: { fetch },
+  } = useSession();
+
+  const [selectedContactIndex, setSelectedContactIndex] = useState(null);
+
+  const closeDrawer = () => setSelectedContactIndex(null);
+  const handleDeleteContact = async () => {
+    const selectedContact = people[selectedContactIndex];
+    await deleteContact(selectedContact, fetch);
+    peopleMutate();
+    closeDrawer();
+  };
+
+  const drawer = (
+    <ContactsDrawer
+      open={selectedContactIndex !== null}
+      onClose={closeDrawer}
+      onDelete={handleDeleteContact}
+    />
+  );
 
-  if (isLoading) return <Spinner />;
   if (addressBookError) return addressBookError;
   if (peopleError) return peopleError;
 
+  const isLoading =
+    (!addressBook && !addressBookError) ||
```

Comment:
true — didn't write the condition, was just switching the order of those lines and didn't amend it (previously it could return the spinner even if there was error). Will fix!